### ISSUE TITLE
Error with path separator character in name

### ISSUE THREAD

**Issue by jdegoes** [quasar-analytics/quasar#1358]
```
[info]     POST should
[error]       x be 400 with fs config and file path in X-File-Name header (13 seconds, 378 ms)
[error]        Falsified after 51 passed tests.
[error]        > ARG_0: FileIn(DirIn(DirIn(Current,DirName(nx.yQq.t)),DirName(0)),FileName(횜gx1톮w2/.mg젮ro.u%fbeanmwv4))
[error]        > ARG_1: DirIn(DirIn(DirIn(Root,DirName(e5f//.fk)),DirName(..)),DirName(IBehG))
[error]        > object[(path,"/e5f$sep$$sep$.fk/$dotdot$/IBehG/nx.yQq.t/0/횜gx1톮w2$sep$.mg젮ro.u%FBeanmwv4")] !== object[(path,"/e5f$sep$$sep$.fk/$dotdot$/IBehG/nx.yQq.t/0/횜gx1톮w2$sep$.mg젮ro.u%fbeanmwv4")] (MountServiceSpec.scala:43)
```

**Comment by jdegoes** [comment]
Seemingly related:

```
[info] Data Service should
[info]   GET
[info]     respond with NotFound
[error]       x if file does not exist
[error]        Falsified after 14 passed tests.
[error]        > ARG_0: FileIn(DirIn(DirIn(Root,DirName(1e)),DirName(g)),FileName(櫈7gZl4q/w%fd))
[error]        > ApiError(404 Path not found.,object[(path,"/1e/g/櫈7gZl4q$sep$w%FD")]) !== ApiError(404 Path not found.,object[(path,"/1e/g/櫈7gZl4q$sep$w%fd")]) (DataServiceSpec.scala:54)
```

**Comment by wemrysi** [comment]
It doesn't look like either of the examples differ due to path separators, however they do appear to differ due to capitalization (`u%FB` vs `u%fb` In the first and `w%FD` vs `w%fd` in the second.

**Comment by mossprescott** [comment]
Maybe the substring `%fd` is getting treated as an escape sequence? We _should_ be URL-encoding the path segments, so not sure what's going on.

**Comment by paulp** [comment]
It's http4s. Hope it's a bug.
```
import org.http4s.util.UrlCodingUtils

scala> UrlCodingUtils.urlEncode("%a")
res26: String = %25a

scala> UrlCodingUtils.urlEncode("%aa")
res27: String = %25AA

scala> UrlCodingUtils.urlEncode("%ag")
res28: String = %25ag
```

**Comment by paulp** [comment]
Fixed in 961dc27908 .
